template_functions.go

```diff
@@ -145,6 +146,35 @@ var (
 		},
 		"slug": Slug,
 		"even": func(a int) bool { return (a % 2) == 0 },
+
+		// Using https://github.com/xeonx/timeago
+		"timeago": func(viewArgs map[string]interface{}, datetime time.Time) string {
+
+			localeStr, ok := viewArgs[CurrentLocaleViewArg].(string)
+			if !ok || strings.Contains(localeStr, "en") {
+				return timeago.English.Format(datetime)
+			}
+			// internalization by current locale
+			var customLanguage = timeago.Config{
```

Comment:
Store the customLanguage outside the function in a map keyed by the localeStr, so that a new timeago.Config object doesnt get created every time a this template function is called